realm/src/androidTest/java/io/realm/RealmQueryTest.java

```diff
@@ -1446,4 +1443,116 @@ public void testIsValidOfRemovedParent() {
         // invalid if parent has been removed
         assertFalse(query.isValid());
     }
+
+
+    public enum SupportedIsEmptyTypes {
+        STRING, BINARY, LIST
+    }
+
+    public enum NotSupportedIsEmptyTypes {
+        INTEGER, FLOAT, DOUBLE, BOOLEAN, OBJECT, DATE
+    }
```

Comment:
It's a bit more cumbersome to write, but this will detect if we add new types and forget to update this unit test. I updated them with this.